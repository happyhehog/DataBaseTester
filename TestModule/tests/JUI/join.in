[@PrintLevel] MAIN
[@Clear]

[@NoOutput]
CREATE TABLE t1(id INT, f1 FLOAT, f2 FLOAT);
CREATE TABLE t2(id INT, f2 FLOAT, f3 FLOAT);
CREATE TABLE t3(id INT, f4 FLOAT, f5 CHAR(50));

INSERT INTO t1 VALUES(1, 3.4, 5.6);
INSERT INTO t1 VALUES(2, 5.6, 5.6);
INSERT INTO t1 VALUES(3, 9, 99);

INSERT INTO t2 VALUES(1, 22, 55);
INSERT INTO t2 VALUES(2, 5.6, 99);
INSERT INTO t2 VALUES(3, 44, 50);

INSERT INTO t3 VALUES(1, 77, 'hello');
INSERT INTO t3 VALUES(2, 66, 'world');
INSERT INTO t3 VALUES(3, 11, 'linux');
[@NoOutput]

@@ Simple join test
SELECT * FROM t1 JOIN t2 ON t1.f2 = t2.f2;

@@ Join with same table and testing as
[@Print]
SELECT * FROM t1 AS t JOIN t1 as t2 on t.id = t2.id;

@@ Join with logic
[@Print]
SELECT * FROM t1
JOIN t2 ON t2.id = 2 or t2.id = 3;

@@ Several joins
[@Print]
SELECT * FROM t1
JOIN t2 ON t1.f2 = t2.f2
JOIN t3 ON t1.id = t3.id;

@@ Left join
[@Print]
SELECT * FROM t1
LEFT JOIN t2 ON t1.f2 = t2.f2;

@@ Right join
[@Print]
SELECT * FROM t1
RIGHT JOIN t2 ON t1.f2 = t2.f2;

@@ Left right join
[@Print]
SELECT * FROM t1
FULL JOIN t2 ON t1.f2 = t2.f2;
