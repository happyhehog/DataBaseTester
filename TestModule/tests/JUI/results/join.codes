{"status":"OK","command":"SELECT * FROM t1 JOIN t2 ON t1.f2 = t2.f2;","answer":"t1.id|t1.f1   |t1.f2   |t2.id|t2.f2   |t2.f3    |\n1    |3.400000|5.600000|2    |5.600000|99.000000|\n2    |5.600000|5.600000|2    |5.600000|99.000000|"}
{"status":"OK","command":"SELECT * FROM t1 AS t JOIN t1 as t2 on t.id = t2.id;","answer":"t.id|t.f1    |t.f2     |t2.id|t2.f1   |t2.f2    |\n1   |3.400000|5.600000 |1    |3.400000|5.600000 |\n2   |5.600000|5.600000 |2    |5.600000|5.600000 |\n3   |9.000000|99.000000|3    |9.000000|99.000000|"}
{"status":"OK","command":"SELECT * FROM t1 JOIN t2 ON t2.id = 2 or t2.id = 3;","answer":"t1.id|t1.f1   |t1.f2    |t2.id|t2.f2    |t2.f3    |\n1    |3.400000|5.600000 |2    |5.600000 |99.000000|\n1    |3.400000|5.600000 |3    |44.000000|50.000000|\n2    |5.600000|5.600000 |2    |5.600000 |99.000000|\n2    |5.600000|5.600000 |3    |44.000000|50.000000|\n3    |9.000000|99.000000|2    |5.600000 |99.000000|\n3    |9.000000|99.000000|3    |44.000000|50.000000|"}
{"status":"OK","command":"SELECT * FROM t1 JOIN t2 ON t1.f2 = t2.f2 JOIN t3 ON t1.id = t3.id;","answer":"t3.id|t3.f4    |t3.f5  |t1.id|t1.f1   |t1.f2   |t2.id|t2.f2   |t2.f3    |\n1    |77.000000|'hello'|1    |3.400000|5.600000|2    |5.600000|99.000000|\n2    |66.000000|'world'|2    |5.600000|5.600000|2    |5.600000|99.000000|"}
{"status":"OK","command":"SELECT * FROM t1 LEFT JOIN t2 ON t1.f2 = t2.f2;","answer":"t1.id|t1.f1   |t1.f2    |t2.id|t2.f2   |t2.f3    |\n1    |3.400000|5.600000 |2    |5.600000|99.000000|\n2    |5.600000|5.600000 |2    |5.600000|99.000000|\n3    |9.000000|99.000000|null |null    |null     |"}
{"status":"OK","command":"SELECT * FROM t1 RIGHT JOIN t2 ON t1.f2 = t2.f2;","answer":"t1.id|t1.f1   |t1.f2   |t2.id|t2.f2    |t2.f3    |\nnull |null    |null    |1    |22.000000|55.000000|\n1    |3.400000|5.600000|2    |5.600000 |99.000000|\n2    |5.600000|5.600000|2    |5.600000 |99.000000|\nnull |null    |null    |3    |44.000000|50.000000|"}
{"status":"OK","command":"SELECT * FROM t1 FULL JOIN t2 ON t1.f2 = t2.f2;","answer":"t1.id|t1.f1   |t1.f2    |t2.id|t2.f2    |t2.f3    |\n1    |3.400000|5.600000 |2    |5.600000 |99.000000|\n2    |5.600000|5.600000 |2    |5.600000 |99.000000|\n3    |9.000000|99.000000|null |null     |null     |\nnull |null    |null     |1    |22.000000|55.000000|\nnull |null    |null     |3    |44.000000|50.000000|"}
