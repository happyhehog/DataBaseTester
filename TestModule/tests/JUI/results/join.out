t1.id|t1.f1   |t1.f2   |t2.id|t2.f2   |t2.f3    |
1    |3.400000|5.600000|2    |5.600000|99.000000|
2    |5.600000|5.600000|2    |5.600000|99.000000|

t.id|t.f1    |t.f2     |t2.id|t2.f1   |t2.f2    |
1   |3.400000|5.600000 |1    |3.400000|5.600000 |
2   |5.600000|5.600000 |2    |5.600000|5.600000 |
3   |9.000000|99.000000|3    |9.000000|99.000000|

t1.id|t1.f1   |t1.f2    |t2.id|t2.f2    |t2.f3    |
1    |3.400000|5.600000 |2    |5.600000 |99.000000|
1    |3.400000|5.600000 |3    |44.000000|50.000000|
2    |5.600000|5.600000 |2    |5.600000 |99.000000|
2    |5.600000|5.600000 |3    |44.000000|50.000000|
3    |9.000000|99.000000|2    |5.600000 |99.000000|
3    |9.000000|99.000000|3    |44.000000|50.000000|

t3.id|t3.f4    |t3.f5  |t1.id|t1.f1   |t1.f2   |t2.id|t2.f2   |t2.f3    |
1    |77.000000|'hello'|1    |3.400000|5.600000|2    |5.600000|99.000000|
2    |66.000000|'world'|2    |5.600000|5.600000|2    |5.600000|99.000000|

t1.id|t1.f1   |t1.f2    |t2.id|t2.f2   |t2.f3    |
1    |3.400000|5.600000 |2    |5.600000|99.000000|
2    |5.600000|5.600000 |2    |5.600000|99.000000|
3    |9.000000|99.000000|null |null    |null     |

t1.id|t1.f1   |t1.f2   |t2.id|t2.f2    |t2.f3    |
null |null    |null    |1    |22.000000|55.000000|
1    |3.400000|5.600000|2    |5.600000 |99.000000|
2    |5.600000|5.600000|2    |5.600000 |99.000000|
null |null    |null    |3    |44.000000|50.000000|

t1.id|t1.f1   |t1.f2    |t2.id|t2.f2    |t2.f3    |
1    |3.400000|5.600000 |2    |5.600000 |99.000000|
2    |5.600000|5.600000 |2    |5.600000 |99.000000|
3    |9.000000|99.000000|null |null     |null     |
null |null    |null     |1    |22.000000|55.000000|
null |null    |null     |3    |44.000000|50.000000|
